# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [7]:
import pandas as pd 
import numpy as np 

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [8]:
np.random.seed(0)

s1 = pd.Series(np.random.randint(1,5,size=100))   # note : increase one to the end value 4>5 and so on 
s2 = pd.Series(np.random.randint(1,4,size=100))
s3 = pd.Series(np.random.randint(10000,30001,size=100))


### Step 3. Let's create a DataFrame by joinning the Series by column

In [9]:
df = pd.concat([s1,s2,s3],axis=1)  # .concat    // or  //  .join ?? errors

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [10]:
df.columns = ['bedrs','bathrs','price_sqr_meter']
df

bedrs  bathrs  price_sqr_meter
0       1       2            15021
1       4       3            11134
2       2       1            26409
3       1       1            19680
4       4       3            27340
..    ...     ...              ...
95      1       2            28161
96      3       2            29745
97      4       1            18338
98      2       1            24676
99      1       1            12562

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [20]:
bigdatacol = pd.concat([s1,s2,s3],ignore_index=True).to_frame(name='bigcolumn')
bigdatacol

bigcolumn
0            1
1            4
2            2
3            1
4            4
..         ...
295      28161
296      29745
297      18338
298      24676
299      12562

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

solu: it is not true 

In [19]:
bigdatacol.shape

(300, 1)

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [21]:
# ignore_index = True 